core/admin/migrations/versions/fc099bd15cbe_.py

```diff
@@ -0,0 +1,65 @@
+"""change default unlimited value for max_users and max_aliases to -1
+
+Revision ID: fc099bd15cbe
+Revises: cd79ed46d9c2
+Create Date: 2019-01-06 13:40:23.372373
+
+"""
+
+# revision identifiers, used by Alembic.
+revision = 'fc099bd15cbe'
+down_revision = 'cd79ed46d9c2'
+
+from alembic import op
+import sqlalchemy as sa
+
+
+domain_table = sa.Table(
+    'domain',
+    sa.MetaData(),
+    sa.Column('max_users', sa.Integer(), nullable=False),
+    sa.Column('max_aliases', sa.Integer(), nullable=False),
+)
+
+
+def upgrade():
+    connection = op.get_bind()
+    connection.execute(
+            domain_table.update().where(
+                domain_table.c.max_users == 0
+                ).values(
+                    max_users=-1
+                )
+            )
+    connection.execute(
+            domain_table.update().where(
+                domain_table.c.max_aliases == 0
+                ).values(
+                    max_aliases=-1
```

Comment:
Can we maybe help users who have accidentally mis-configured their domains by forcing everything `<=0` to `-1`? Although then the downgrade path wouldn’t be clean anymore …. And maybe the user would be surprised since while he accidentally entered a negative number, he didn’t necessarily *mean* infinity?
Oh well, is hard — I guess staying as deterministic like this is the best after all …

Reply:
Yes, I had this thought too.  But then I thought that maybe previously someone wanted to disable aliases for a domain (`0` was infinity, and e.g. -5 disabled aliases). 
And also the use case you mentioned is against the `<=0` migration. So I think you're right, we should keep the way it is ...
Thanks for the additional input!